### PR TITLE
ICMSLST-379 Add Deactivated Firearms Licence Application 

### PR DESCRIPTION
**Pull request by MattHolmes123**
Other changes:

- Refactor ImportApplicationTypes types and subtypes to use TextChoices.

### CODE REVIEW

File: web/domains/case/_import/models.py [hunk 1]
```diff
@@ -14,24 +14,19 @@
 
 
 class ImportApplicationType(models.Model):
-    TYPE_FIREARMS_AMMUNITION_CODE = "FA"
-    TYPE_SANCTION_ADHOC = "SAN_ADHOC_TEMP"  # TODO: missing from legacy data extract, hence TEMP
-    TYPE_WOOD_QUOTA = "WD"
-    TYPE_DEGROGATION_FROM_SANCTIONS_BAN = "SAN"
+    class AppTypes(models.TextChoices):
```
Comment (oskusalerma) [inline web/domains/case/_import/models.py:17]: as discussed offline:

- enum names should be uppercase
- apptypes -> Types
- AppSubTypes -> SubTypes
- shorten names (e.g. firearms_ammunition_code -> firearms, etc)

File: web/domains/case/_import/models.py [hunk 2]
```diff
@@ -14,24 +14,19 @@
...
+        firearms_ammunition_code = ("FA", "Firearms and Ammunition")
+        # TODO: missing from legacy data extract, hence TEMP
+        sanction_adhoc = ("SAN_ADHOC_TEMP", "Sanctions and Adhoc")
+        wood_quota = ("WD", "Wood (Quota)")
+        degrogation_from_sanctions_ban = ("SAN", "Derogation from Sanctions Import Ban")
```
Comment (oskusalerma) [inline web/domains/case/_import/models.py:22]: fix the typo while you're at it (degrogation -> derogation)

**Review by oskusalerma — CHANGES_REQUESTED** [review]

**Review by oskusalerma — APPROVED** [review]

File: web/domains/case/_import/fa_dfl/models.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from web.models import ImportApplication
+
+
+class DFLApplication(ImportApplication):
+    """Firearms & Ammunition Deactivated Firearms Licence application class"""
```
Comment (oskusalerma) [inline web/domains/case/_import/fa_dfl/models.py:5]: you don't have to say class

File: web/domains/case/_import/views.py [hunk 4]
```diff
@@ -41,40 +42,75 @@ class ImportApplicationChoiceView(TemplateView, PermissionRequiredMixin):
...
+        import_application_type,  # type: ignore[arg-type]
+        model_class,
+        redirect_view,
+    )
+
+
+@login_required
+@permission_required("web.importer_access", raise_exception=True)
+def create_firearms_dfl(request: HttpRequest) -> HttpResponse:
+    """Create a Firearms & Ammunition - Deactivated Firearms Licence application."""
```
Comment (oskusalerma) [inline web/domains/case/_import/views.py:90]: not sure this comment is adding any value

File: web/migrations/0060_add_dflapplication.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+# Generated by Django 3.1.8 on 2021-05-07 07:37
```
Comment (oskusalerma) [inline web/migrations/0060_add_dflapplication.py:1]: 60 is already used

**Review by oskusalerma — CHANGES_REQUESTED** [review]

File: web/migrations/0060_add_dflapplication.py (hunk 5 above)
Comment (MattHolmes123) [reply web/migrations/0060_add_dflapplication.py:1]: Ahh yer just rebased and got 60

**Review by kevinetienne — APPROVED** [review]

**Review by oskusalerma — APPROVED** [review]

**Pull request merged**